tests/unit/moc_openshift/test_moc_openshift_project.py

```diff
@@ -45,3 +47,56 @@ def test_create_project(moc):
 def test_delete_project(moc):
     moc.delete_project("fake-project")
     moc.client.resources.get.return_value.delete.assert_called_with(name="fake-project")
+
+
+def test_get_users_in_project_with_nonexistent_project(moc):
+    moc.client.resources.get.return_value.get.side_effect = kexc.NotFoundError(
+        mock.Mock()
+    )
+
+    with pytest.raises(kexc.NotFoundError):
+        moc.get_users_in_project("nonexistent_project")
+
+def test_get_users_in_project_with_no_rolebindings(moc):
+    dummy_project = {"name": "project1"}
+    moc.get_project = mock.Mock(return_value=dummy_project)
+
+    moc.get_rolebindings = mock.Mock(side_effect=kexc.NotFoundError(mock.Mock()))
+
+    users = moc.get_users_in_project("project1")
+
+    assert users == []
+
+def test_get_users_in_project_with_project_with_one_rolebinding(moc):
+    dummy_project = {"name": "project1"}
+    moc.get_project = mock.Mock(return_value=dummy_project)
+
+    def get_rolebindings_side_effect(project_name, role):
+        if role == "view":
+            return {"role": "view", "subjects": [{"kind": "User", "name": "viewer"}]}
+        else:
+            raise kexc.NotFoundError(mock.Mock())
+
+    moc.get_rolebindings = mock.Mock(side_effect=get_rolebindings_side_effect)
+
+    users = moc.get_users_in_project("project1")
+
+    assert users == ["viewer"]
+
+def test_get_users_in_project_with_multiple_rolebindings(moc):
+    dummy_project = {"name": "project1"}
+    moc.get_project = mock.Mock(return_value=dummy_project)
+
+    def get_rolebindings_side_effect(project_name, role):
+        if role == "admin":
+            return {"role": "admin", "subjects": [{"kind": "User", "name": "administrator"}]}
+        elif role == "view":
+            return {"role": "view", "subjects": [{"kind": "User", "name": "viewer"}]}
+        elif role == "edit":
+            return {"role": "edit", "subjects": [{"kind": "User", "name": "editor"}]}
+
+    moc.get_rolebindings = mock.Mock(side_effect=get_rolebindings_side_effect)
+
+    users = moc.get_users_in_project("project1")
+
+    assert set(users) == set(["viewer", "editor", "administrator"])
```

Comment:
You can simplify `get_rolebindings_side_effect` a bit by just generating the user names from the role name. You also probably want to handle the case of the role not matching any of the conditions. E.g:

```suggestion
    # pylint: disable=unused-argument
    def get_rolebindings_side_effect(project_name, role):
        if role in ["admin", "view", "edit"]:
            return {"role": role, "subjects": [{"kind": "User", "name": f"{role}-user"}]}

        raise ValueError(role)

    moc.get_rolebindings = mock.Mock(side_effect=get_rolebindings_side_effect)

    users = moc.get_users_in_project("project1")

    assert set(users) == set(["view-user", "admin-user", "edit-user"])
```

Reply:
Thanks @larsks, this is better. Also, I completely forgot I could disable some pylint errors, so thanks for pointing that out!

Reply:
Not important here, but you can also avoid the "unused argument" error by using `_` as the argument name:

```
    def get_rolebindings_side_effect(_, role):
        if role in ["admin", "view", "edit"]:
            return {"role": role, "subjects": [{"kind": "User", "name": f"{role}-user"}]}
```

Naming a variable `_` tells pylint, "I don't care about this".

Reply:
> You also probably want to handle the case of the role not matching any of the conditions.

Not sure how to trigger that, because `get_rolebindings` is called with only the roles defined in `OPENSHIFT_ROLES`.

Reply:
I wasn't expecting to trigger it; I just felt uncomfortable with the function returning `None` if we made a typo somewhere.

Reply:
I think I understand what you mean now, and that's why you included `raise ValueError(role)` so it's clear if we goof up elsewhere. Thanks. I have included your suggestions, though not directly with the "commit suggestion" button because I was already fixing other things.